File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        Set<Integer> numsSet = new HashSet<>();
        for(int num: nums) {
            numsSet.add(num);
        }
        ListNode tempHead = new ListNode();
        tempHead.next = head;
        ListNode temp = tempHead;

        while(temp.next != null) {
            if(numsSet.contains(temp.next.val)) {
                ListNode next = temp.next.next;
                temp.next = temp.next.next;
            } else {
                temp = temp.next;
            }
        }

        return tempHead.next;
    }
}

File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        temp = ListNode()
        tempHead = temp
        nums.sort()
        length = len(nums)

        def is_present(num):
            l = 0
            r = length - 1

            while l <= r:
                m = (r + l) // 2
                if nums[m] == num:
                    return True
                if nums[m] > num:
                    r = m - 1
                else:
                    l = m + 1

            return False        

        while head != None:
            if not is_present(head.val):
                temp.next = head
                head = head.next
                temp = temp.next
                temp.next = None
            else:
                head = head.next

        return tempHead.next        